custom_components/tuya_local/devices/beok_tol47_thermostat.yaml

```diff
@@ -0,0 +1,192 @@
+name: Beok TOL47 thermostat
+primary_entity:
+  entity: climate
+  dps:
+    - id: 1
+      type: boolean
+      name: power
+      mapping:
+        - dps_val: false
+          value: "off"
+      hidden: true
+    - id: 2
+      type: integer
+      name: temperature
+      mapping:
+        - scale: 10
+          step: 5
+    - id: 3
+      type: integer
+      name: current_temperature
+      mapping:
+        - scale: 10
+    - id: 4
+      type: string
+      name: preset_mode
+      mapping:

```

Comment:
This seems like a strange mapping.  Are you sure you got preset_mode and hvac_mode around the right way?

Reply:
Yes, a little bit strange for me too. Debugged dps are:

```
"device_config": {
"friendly_name": "Living Room Thermostat",
"local_key": "xxxxxxxxxxxxxxx",
"host": "192.168.0.55",
"device_id": "xxxxxxxxxxxxxxx",
"protocol_version": "3.3",
"scan_interval": 1,
"dps_strings": [
	"1 (value: True)",	# Switch
	"2 (value: 205)",	# Set Temperature *10
	"3 (value: 202)", 	# Current Temperature *10
	"4 (value: cold)",	# Mode "cold": Manual "hot": Program "wind": Temporary Program 
	"5 (value: 1)",		# Working status 0: off 1: on
	"8 (value: False)",	# Child lock
	"13 (value: True)",	# Sound
	"16 (value: 0)",	# Fault alarm
	"20 (value: -7)",	# Temperature calibration *10
	"21 (value: 30)",	# Max setting temperature
	"25 (value: 1)",        # Sensor selection 1: internal 2: external 3: both
	"26 (value: False)",    # Frost protection
	"31 (value: False)",	# Reset
	"41 (value: 3)",       	# Backlight brightness 0: Close 1: Low 2: Middle 3: High
	"42 (value: 1)",      	# Working day settings 1: 5+2 2: 6+1 3: 7
 	"101 (value: False)",	# Output main reverse
 	"105 (value: 5)",     	# Temperature control switch difference *10
 	"107 (value: 30)"       # External Sensor temperature limit
```
- on dp4, i can change value with the Tuya Smart app "Mode" settings, and if i change it from HA, the app settings reflect correctly.
- on dp5, value 0 if dry contact is open, 1 if closed (=boyler heating)
        		
Or i completely wrong interpreting preset_mode and hvac_mode?

Reply:
In that case, preset_mode is correct for dp 4, but I would use hvac_action for dp 5, and hvac_mode directly for dp 1.